examples/java-matter-controller/java/src/com/matter/controller/commands/common/FutureResult.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+// FutureResult encapculates the optional realResult, application would wait for realResult set by
+// other thread wben receiving
+// data from the other end.
+// If the expected duration elapsed without receiving the expected realResult, the runtime exception
+// would be raised.
+package com.matter.controller.commands.common;
+
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class FutureResult implements Result {
+  private Optional<RealResult> realResult;
+  private long timeoutMs = 0;
+  private static Logger logger = Logger.getLogger(FutureResult.class.getName());
+
+  public void setTimeoutMs(long timeoutMs) {
+    this.timeoutMs = timeoutMs;
+  }
+
+  public synchronized void setRealResult(RealResult realResult) {
+    if (this.realResult.isPresent()) {
+      throw new RuntimeException("error, real result has been set!");
+    }
+    this.realResult = Optional.of(realResult);
+    notify();
+  }
+
+  public synchronized boolean getResult() {
+    long start = System.currentTimeMillis();
+    while (!realResult.isPresent()) {
+      try {
+        if (System.currentTimeMillis() > (start + timeoutMs)) {
+          throw new RuntimeException("timeout!");
+        }
+        wait();
+      } catch (InterruptedException e) {
+      }
+    }
+
+    if (!realResult.get().getResult()) {
+      logger.log(Level.INFO, "error: %s", realResult.get().getError());
+      throw new RuntimeException("received failure test result");
+    }
+
+    return realResult.get().getResult();
```

Comment:
This can only be 'true' right?

In this case, why do we even have getResult() returning a bool since we seem to only ever either throw exception or return true.

Reply:
just rename it to waitResult, removed the boolean return type. thanks